File: Mario/mario.c
// Abstraction and scope

#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Prompt user for height
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while(n < 1 || n > 8);
    
    // Print the pyramid
    for(int i = 0; i < n; i++)
    {
        // Print spaces before blocks
        for(int j = 0; j < n - 1 - i; j++)
        {
            printf(" ");
        }
        
        //Print first set of blocks
        for(int j = 0; j < i + 1; j++)
        {
            printf("#");
        }
        
        //Print gap
        printf("  ");
        
        //Print second set of blocks
        for(int j = 0; j < i + 1; j++)
        {
            printf("#");
        }
        printf("\n");
    }               
}

File: mario.py
# Print a staircase of blocks sized by user input
from cs50 import get_int


def main():
    height = get_positive_int("Height: ")
    build(height)


# Prompt user for height
def get_positive_int(prompt):
    while True:
        n = get_int(prompt)
        if n > 0 and n < 9:
            return n

# Print spaces and blocks
def build(height):
    for x in range(height):

        for y in range(height - 1 - x):
            print(" ", end='')

        for y in range(x + 1):
            print("#", end='')

        print()


if __name__ == "__main__":
    main()
